core/src/test/scala/com/databricks/labs/remorph/parsers/common/FunctionBuilderSpec.scala

```diff
@@ -191,243 +191,248 @@ class FunctionBuilderSpec extends AnyFlatSpec with Matchers with TableDrivenProp
+      ("SIGN", FunctionDefinition.standard(1)),
+      ("SIN", FunctionDefinition.standard(1)),
+      ("SMALLDATETIMEFROMPARTS", FunctionDefinition.standard(5)),
+      ("SOUNDEX", FunctionDefinition.standard(1)),
+      ("SPACE", FunctionDefinition.standard(1)),
+      ("SQL_VARIANT_PROPERTY", FunctionDefinition.standard(2)),
+      ("SQRT", FunctionDefinition.standard(1)),
+      ("SQUARE", FunctionDefinition.standard(1)),
+      ("STATS_DATE", FunctionDefinition.standard(2)),
+      ("STDEV", FunctionDefinition.standard(1)),
+      ("STDEVP", FunctionDefinition.standard(1)),
+      ("STR", FunctionDefinition.standard(1, 3)),
+      ("STRING_AGG", FunctionDefinition.standard(2, 3)),
+      ("STRING_ESCAPE", FunctionDefinition.standard(2)),
+      ("STUFF", FunctionDefinition.standard(4)),
+      ("SUBSTR", FunctionDefinition.standard(2, 3)),
+      ("SUBSTRING", FunctionDefinition.standard(2, 3)),
+      ("SUM", FunctionDefinition.standard(1)),
+      ("SUSER_ID", FunctionDefinition.standard(0, 1)),
+      ("SUSER_NAME", FunctionDefinition.standard(0, 1)),
+      ("SUSER_SID", FunctionDefinition.standard(0, 2)),
+      ("SUSER_SNAME", FunctionDefinition.standard(0, 1)),
+      ("SWITCHOFFSET", FunctionDefinition.standard(2)),
+      ("SYSDATETIME", FunctionDefinition.standard(0)),
+      ("SYSDATETIMEOFFSET", FunctionDefinition.standard(0)),
+      ("SYSTEM_USER", FunctionDefinition.standard(0)),
+      ("SYSUTCDATETIME", FunctionDefinition.standard(0)),
+      ("TAN", FunctionDefinition.standard(1)),
+      ("TIMEFROMPARTS", FunctionDefinition.standard(5)),
+      ("TODATETIMEOFFSET", FunctionDefinition.standard(2)),
+      ("TOSTRING", FunctionDefinition.standard(0)),
+      ("TRANSLATE", FunctionDefinition.standard(3)),
+      ("TRIM", FunctionDefinition.standard(1, 2)),
+      ("TYPE_ID", FunctionDefinition.standard(1)),
+      ("TYPE_NAME", FunctionDefinition.standard(1)),
+      ("TYPEPROPERTY", FunctionDefinition.standard(2)),
+      ("UNICODE", FunctionDefinition.standard(1)),
+      ("UPPER", FunctionDefinition.standard(1)),
+      ("USER", FunctionDefinition.standard(0)),
+      ("USER_ID", FunctionDefinition.standard(0, 1)),
+      ("USER_NAME", FunctionDefinition.standard(0, 1)),
+      ("VALUE", FunctionDefinition.xml(2)),
+      ("VAR", FunctionDefinition.standard(1)),
+      ("VARP", FunctionDefinition.standard(1)),
+      ("XACT_STATE", FunctionDefinition.standard(0)),
+      ("YEAR", FunctionDefinition.standard(1)))
 
     val functionBuilder = new TSqlFunctionBuilder
 
-    forAll(functions) { (functionName: String, expectedArity: Option[FunctionDefinition]) =>
-      functionBuilder.functionDefinition(functionName) shouldEqual expectedArity
+    forAll(functions) { (functionName: String, expectedArity: FunctionDefinition) =>
+      {
+        val actual = functionBuilder.functionDefinition(functionName)
+        actual.nonEmpty shouldEqual true
+        actual.get.arity shouldEqual expectedArity.arity
```

Comment:
Minor nit, I think this will be more understandable if it fails.
```suggestion
        actual.map(_.arity) shouldEqual Some(expectedArity.arity)
```

Should we also be checking the function-type?
```suggestion
        actual.map(_.functionType) shouldEqual Some(StandardFunction)
```

Reply:
No we shouldn't! The test is called "FunctionBuilder" should "return correct arity for each function"
It was incorrectly comparing functions. Now it does what it says!

Reply:
@ericvergnaud: thanks for the clarification; I understand the name of the test suggests this is the intent and that PR change is consistent with this.

However previously we _were_ checking the `functionType` and it's not clear to me that we're now checking it anywhere. I do see some adjacent tests for `functionType` on the TSQL side of things, but not on the SnowFlake side here. Have we lost something here unintentionally? Are we now missing a test?

Reply:
IMHO this test is completely useless. It smells like testing java getters.

Reply:
What we want to test is that a function behaves as expected, not that its contract complies with yet another spec.
But removing the test should go in another PR, if we all agree.

Reply:
And we should add tests for each function. Many other PRs.

Reply:
> @ericvergnaud: thanks for the clarification; I understand the name of the test suggests this is the intent and that PR change is consistent with this.
> 
> However previously we _were_ checking the `functionType` and it's not clear to me that we're now checking it anywhere. I do see some adjacent tests for `functionType` on the TSQL side of things, but not on the SnowFlake side here. Have we lost something here unintentionally? Are we now missing a test?

I may have added the TSQL stuff and not added it for Snowflake because I did all the amalgamation of function calls into one or two grammar rules there first, then we did it in Snowflake after the fact. I think you are correct here.

Reply:
So we're calling code without checking its behavior only to improve code coverage ratio ?

Reply:
Getting back to the changes, I understand the situation now. Thanks for the clarifications.